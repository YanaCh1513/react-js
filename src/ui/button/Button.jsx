import { PropTypes } from "@mui/material"

export function Button(props) {
    return (
        <>
            <button {...props}>{props.children}</button>
        </>
    )
}

Button.propTypes = {
    type: PropTypes.string
}