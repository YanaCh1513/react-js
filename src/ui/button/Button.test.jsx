import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'


import { Button } from './Button'

describe('Button testing', () => {
    it('Render Button component', () => {
        render(<Button>Ok</Button>)
    })
    it('Render Button with snapshot', () => {
        const { asFragment } = render(<Button>Ok</Button>)
        expect(asFragment()).toMatchSnapshot()
    })
    it('Render Button with text "Ok"', () => {
        render(<Button>Ok</Button>)
        expect(screen.getByText(/Ok/)).toBeInTheDocument()
    })
    it('Render multiple Button', () => {
        render(
            <>
                <Button>Edit</Button>
                <Button>Delete</Button>
            </>
        )
        expect(screen.queryAllByRole('button').length).toBe(2)
    })
    it('Render Button is disabled', () => {
        render(<Button disabled>Delete</Button>)
        expect(screen.getByText(/Delete/)).toBeDisabled()
    })
    // needed to refresh snapshot: npm run test -- -u
    it('Render Button with style "color = black"', () => {
        render(<Button>Delete</Button>)
        expect(screen.getByText(/Delete/)).toHaveStyle({ color: "black" })
    })

    it('Button event "click"', async () => {
        const mockHandler = jest.fn()
        render(<Button onClick={mockHandler}>Delete</Button>)
        await userEvent.click(screen.getByText(/Delete/))
        expect(mockHandler).toHaveBeenCalledTimes(1)
    })

    it('Checkbox test', async () => {
        const onChange = jest.fn()
        render(<input type="checkbox" onChange={onChange}></input>)
        const cbx = screen.getByRole('checkbox')
        await userEvent.dblClick(cbx)
        expect(onChange).toHaveBeenCalledTimes(2)
        expect(cbx).not.toBeChecked()
    })
}) 