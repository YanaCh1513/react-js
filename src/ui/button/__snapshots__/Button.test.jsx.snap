// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing Render Button with snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: black;"
  >
    Ok
  </button>
</DocumentFragment>
`;
